pose_conversions.py

```diff
@@ -0,0 +1,262 @@
+"""Example to show conversions to/from Transformation Matrix.
+
+Zivid primarily operate with a (4x4) Transformation Matrix (Rotation Matrix + Translation Vector).
+This example shows how to use Eigen to convert to and from:
+  AxisAngle, Rotation Vector, Roll-Pitch-Yaw, Quaternion
+
+ It provides convenience functions that can be reused in applicable applications.
+"""
+
+import enum
+from pathlib import Path
+from dataclasses import dataclass, field
+import numpy as np
+import cv2
+from scipy.spatial.transform import Rotation as R
+
+
+def _main():
+    np.set_printoptions(precision=4)
+    print_header(f"This example shows conversions to/from Transformation Matrix")
+
+    transformation_matrix = get_transformation_matrix_from_yaml("robotTransform.yaml")
+    print(f"Transformation Matrix:\n{transformation_matrix}")
+
+    # Extract Rotation Matrix and Translation Vector from Transformation Matrix
+    print(f"Rotation Matrix:\n{transformation_matrix[:-1,:-1]}")
+    print(f"Translation Vector:\n{transformation_matrix[:-1, -1]}")
+
+    ###
+    # Convert from Zivid to Robot (Transformation Matrix --> any format)
+    ###
+    print_header("Convert from Zivid to Robot")
+    axis_angle = zivid_to_axis_angle(transformation_matrix)
+    print(f"AxisAngle:\n{axis_angle.angle}, {axis_angle.axis}")
+
+    rotation_vector = zivid_to_rotation_vector(transformation_matrix)
+    print(f"Rotation Vector:\n{rotation_vector}")
+
+    quaternion = zivid_to_quaternion(transformation_matrix)
+    print(f"Qaternion:\n{quaternion}")
+
+    rpy_list = list()
+    rotation_matrix = transformation_matrix[:3, :3]
+    rotation = R.from_matrix(rotation_matrix)
```

Comment:
Perhaps replace 43 and 44 with rotation = R.from_matrix(transformation_matrix[:3, :3])?

Reply:
solved through refactor